core/include/masks/annulus2.hpp

```diff
@@ -0,0 +1,168 @@
+     *  e.g. used for the itk strip endcaps.
+     * 
+     * It is defined by the two radii _values[0] and  _values[1] in the polar
+     * coordinate system of and endcap strip module, as well as the two phi 
+     * boundaries, _values[2] and _values[3], that are only easily definable in 
+     * the local strips system (relative to the average Phi along the 
+     * strips r coordinate). Using a conversion between the two coordinate 
+     * systems, these boundaries can be checked with a tolerance in r (t0 and 
+     * t1), as well as phi (t3 and t3). 
+     * Due to the local polar coordinate system of the strips needing a 
+     * different origin from the discs polar one, three additional conversion
+     * parameters are included (_values[4], values[5], _values[6]).
+     * Here, the first two are the origin shift in xy, while _values[6] is the 
+     * average Phi angle mentioned above.
+     **/
+    template <typename scalar_type,
+              typename intersector_type = planar_intersector,
+              typename links_type = bool,
+              unsigned int kMaskIdentifier = e_annulus2>
+    struct annulus2
+    {
+        using mask_values = darray<scalar_type, 7>;
+
+        mask_values _values = {0., std::numeric_limits<scalar_type>::infinity(),
+                               -std::numeric_limits<scalar_type>::infinity(), 
+                               std::numeric_limits<scalar_type>::infinity(),
+                               0., 0., 0.};
+
+        links_type _links;
+
+        static constexpr unsigned int mask_identifier = kMaskIdentifier;
+
+        /** Assignment operator from an array, convenience function
+         * 
+         * @param rhs is the right hand side object
+         **/
+        annulus2<scalar_type, intersector_type, links_type, kMaskIdentifier> &
+        operator=(const darray<scalar_type, 7> &rhs)
+        {
+            _values = rhs;
+            return (*this);
+        }
+
+        /** Mask operation 
+         * 
+         * @tparam point2_type is the type of the point to be checked w.r.t. to
+         * the mask bounds
+         * 
+         * @param p the point to be checked in local polar coord
+         * @param t0 is the tolerance in minR
+         * @param t1 is the tolerance in minPhi
+         * 
+         * @return an intersection status e_inside / e_outside
+         **/
+        template<typename local_type>
+        intersection_status is_inside(const typename local_type::point2 &p,
+                                       scalar_type t0 = std::numeric_limits<scalar_type>::epsilon(),
+                                       scalar_type t1 = std::numeric_limits<scalar_type>::epsilon()) const
+        {
+           // The two quantities to check: r^2 in module system, phi in strips system
+
+           // In cartesian coordinates go to modules system by shifting origin
+           if (typeid(local_type) == typeid(__plugin::cartesian2)) {
```

Comment:
For my education: is this supposed to be a runtime check? I think it could be a compile-time if.

Reply:
Oh, then, I think we should change it to compile time.

The way the masks work, it should always be known at compile time.

Reply:
```cpp
if constexpr(std::is_same_v<local_type, __plugin::cartesian2>) {
// ...
}
```

Reply:
I've had look at this and cppreference says about typeid operator:
"When applied to an expression of polymorphic type, evaluation of a typeid expression may involve runtime overhead (a virtual table lookup), otherwise typeid expression is resolved at compile time."
Also:
"In all cases, top-level cv-qualifiers are ignored by typeid (that is, typeid(const T) == typeid(T))."
which is what we want, right?

Reply:
Yeah I think you're right that `typeid` resolves at compile-time, but the if I believe is emitted as machine code. It's possible that the compiler removes it as an optimization, like it probably does `if(true)`, but as far as I know `if constexpr` guarantees this.

Reply:
I had a look at godbolt, on GCC10, starting from `-O1`, this is contracted to the correct branch. So it's definitely not a huge deal. However, my personal opinion is that the guarantee of `if constexpr` is still preferrable.

Reply:
It seems typeids equals operator does not like constexpr, so I went with your solution, thanks!